src/components/DateSelect/DateSelect.jsx

```diff
@@ -201,6 +216,30 @@ const DateSelect = createClass({
 		this.initialMonth = new Date(this.props.initialMonth);
 	},
 
+	componentDidMount() {
+		const {
+			isFontSizeRelative,
+			monthsShown,
+		} = this.props;
+
+		if (isFontSizeRelative) {
+			const rootElement = ReactDOM.findDOMNode(this.rootRef);
+			const {
+				width,
+				height,
+			} = rootElement.getBoundingClientRect();
+			const navButtonsWidth = NAV_BUTTON_SIZE * 2;
+			const oneMonthShownWidth = (width - navButtonsWidth) / monthsShown + navButtonsWidth;
+			const size = Math.sqrt(oneMonthShownWidth * height);
+			const relativeFontSize = Math.round(size / 24);
+			const relativeMinWidth = (((width - navButtonsWidth) / monthsShown) * 10.1075 / relativeFontSize) * monthsShown + navButtonsWidth;
+
+			rootElement.style.fontSize = `${relativeFontSize}px`;
```

Comment:
it's documented in the prop as only affecting the _initial_ font-size so i think it's ok. i'd rather not have to recompute onresize and on props change